### PR TITLE
Sb 28285 template

### PR DESCRIPTION
**Pull request by vishwanath1004**

### CODE REVIEW

File: src/app/manage-learn/project/project-templateview/project-templateview.page.ts [hunk 1]
```diff
@@ -107,34 +118,56 @@ export class ProjectTemplateviewPage {
...
-  async sortTasks() {
-    let projectData: any = await this.utils.getSortTasks(this.project);
-    this.project = projectData.project;
-    this.sortedTasks = projectData.sortedTasks;
-    this.taskCount = projectData.taskCount;
+  async getTemplateByExternalId() {
+      const config = {
+      url: urlConstants.API_URLS.PROJECT_TEMPLATE_DETAILS + this.id,
+    }
+    this.unnatiService.get(config).subscribe(success => {
```
Comment (kiranharidas187) [inline src/app/manage-learn/project/project-templateview/project-templateview.page.ts:128]: Put the API call in sevice

File: src/app/manage-learn/shared/components/task-card/task-card.component.ts [hunk 2]
```diff
@@ -12,15 +17,145 @@ export class TaskCardComponent implements OnInit {
...
 
   }
+  async startAssessment(task) {
+    if (!this.networkFlag) {
+      this.toast.showMessage('FRMELEMNTS_MSG_YOU_ARE_WORKING_OFFLINE_TRY_AGAIN', 'danger');
+      return;
+    }
+    let payload = await this.utils.getProfileInfo();
+     const config = {
+       url: urlConstants.API_URLS.START_ASSESSMENT + `${this.data._id}?taskId=${task._id}`,
```
Comment (kiranharidas187) [inline src/app/manage-learn/shared/components/task-card/task-card.component.ts:48]: Put the network-related and API call in project service

File: src/app/manage-learn/shared/components/task-card/task-card.component.ts [hunk 3]
```diff
@@ -12,15 +17,145 @@ export class TaskCardComponent implements OnInit {
...
+         });
+       },
+       (error) => {
+         this.toast.showMessage(this.allStrings["FRMELEMNTS_MSG_CANNOT_GET_PROJECT_DETAILS"], "danger");
+       }
+     );
+ }
+
+ getProjectTaskStatus() {
+  if (!this.data.tasks && !this.data.tasks.length) {
```
Comment (kiranharidas187) [inline src/app/manage-learn/shared/components/task-card/task-card.component.ts:96]: This API call is already handled on the Details Page,, don't put it here. If there are 2 observation type task we will have to call the API 2 times as per this logic. Which is not a good approach

File: src/app/manage-learn/shared/components/task-card/task-card.component.ts [hunk 4]
```diff
@@ -12,15 +17,145 @@ export class TaskCardComponent implements OnInit {
...
+        return;
+      }
+      this.updateAssessmentStatus(success.result);
+    },
+    (error) => {
+    }
+  );
+ }
+
+ updateAssessmentStatus(data) {
```
Comment (kiranharidas187) [inline src/app/manage-learn/shared/components/task-card/task-card.component.ts:125]: this same is handled in task details I guess. Please heck once

Comment (vishwanath1004) [reply src/app/manage-learn/shared/components/task-card/task-card.component.ts:125]: okay bro will check

File: src/app/manage-learn/shared/components/task-card/task-card.component.ts (hunk 3 above)
Comment (vishwanath1004) [reply src/app/manage-learn/shared/components/task-card/task-card.component.ts:96]: Okay

File: src/app/manage-learn/shared/components/task-card/task-card.component.ts [hunk 5]
```diff
@@ -12,15 +15,31 @@ export class TaskCardComponent implements OnInit {
...
 statuses =statusType;
-  constructor(
-    private router: Router
-  ) { }
+networkFlag: boolean;
+private _networkSubscription: Subscription;
+allStrings;
+  constructor( private router : Router,
+    private commonUtilService : CommonUtilService,
+    private projectService :ProjectService ) {
```
Comment (kiranharidas187) [inline src/app/manage-learn/shared/components/task-card/task-card.component.ts:23]: Do we need this now?

File: src/app/manage-learn/project/project-templateview/project-templateview.page.html [hunk 6]
```diff
@@ -1,3 +1,12 @@
+<ion-header class="ion-no-border" class="header" translucent="true">
```
Comment (kiranharidas187) [inline src/app/manage-learn/project/project-templateview/project-templateview.page.html:1]: Why do we need the new header here

File: src/app/manage-learn/project/project-templateview/project-templateview.page.ts [hunk 7]
```diff
@@ -175,4 +204,13 @@ export class ProjectTemplateviewPage {
     //   // go to login page
     // }
   }
+
+  async closeTemplate() {
```
Comment (kiranharidas187) [inline src/app/manage-learn/project/project-templateview/project-templateview.page.ts:208]: Do we need this code?

**Pull request merged**